app/services/proceeding_type_filter.rb

```diff
@@ -40,7 +40,10 @@ def configure_plf_proceedings!
     return unless current_proceedings_have_plf
 
     @results.delete_if do |proceeding|
-      proceeding["ccms_code"] !~ /^PBM/ # if already has an PLF proceeding, exclude all non-PLF proceedings
+      [
+        proceeding["ccms_code"].match?("^(?!PBM)"), # if already has an PLF proceeding, exclude all non-PLF proceedings
+        remove_plf_for_means_tested(proceeding),
+      ].any?
```

Comment:
naming feels confusing: "delete if...remove plf for-means tested".  I think the method and its use would be more clearly reflected by `has_means_test_plf_mismatch?(proceeding)` or similar

```ruby
    @results.delete_if do |proceeding|
      proceeding["ccms_code"] !~ /^PBM/ # if already has an PLF proceeding, exclude all non-PLF proceedings
      [
        proceeding["ccms_code"].match?("^(?!PBM)"), # if already has an PLF proceeding, exclude all non-PLF proceedings
        has_means_test_plf_mismatch?(proceeding),
      ].any?
```
Could also extract the other to a method  so

```ruby
    @results.delete_if do |proceeding|
      [
        is_not_plf?(proceeding)
        has_means_test_plf_mismatch?(proceeding),
      ].any?

Reply:
#NamingThingsIsHard 👍